### PR TITLE
add PanicError and Recover() helpers

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

File: panic.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import "fmt"
+
+// PanicError is a caught error from a panic() call
+type PanicError interface {
+	Error() string
+	Recovered() interface{}
+}
+
+// Recover attempts to recover from a panic
+func Recover() PanicError {
```
Comment (karasz) [inline panic.go:12]: recover() called outside defer is funky, might not do what we expect

File: panic.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+		// pass through
+		return p
+	} else {
+		// wrap it
+		return &panicError{rvr}
+	}
+}
+
+type panicError struct {
+	payload interface{}
```
Comment (karasz) [inline panic.go:26]: `recover()` returns `any`, is there any reason to keep using `interface{}`?

File: panic.go (hunk 1 above)
Comment (amery) [reply panic.go:12]: `slog.Recover()` is intended to be used inside a defer.

```go
defer func() {
   if err := slog.Recover(); err != nil {
      ...
   }
}()
```

File: panic.go (hunk 2 above)
Comment (amery) [reply panic.go]: amended

**Comment by amery** [comment]
@karasz amended

**Comment by amery** [comment]
v0.4.1 tagged

**Pull request merged**